### PR TITLE
Fix:Static pod status is always Init:0/1 if unable to get init contai…

### LINKED ISSUE #108537: Static pod status is always Init:0/1 when init container GC'd before kubelet restart.
### What happened?

When init container of static pod was GC'd and kubelet was restarted. The status of the static pod is alway Init:0/1 and never come back to Running.

```shell

kube-controller-manager-paas-192-168-16-112     0/1     Init:0/1                0          11d     192.168.16.112   paas-192-168-16-112   <none>           <none>
kube-controller-manager-paas-192-168-16-113     1/1     Running                 1          11d     192.168.16.113   paas-192-168-16-113   <none>           <none>

```

### What did you expect to happen?

The pod status is Running

### How can we reproduce it (as minimally and precisely as possible)?

1. remove exited init contianer of one static pod.
2. restart kubelet

### Anything else we need to know?

I noticed #96572 makes containers not restarting after init container GC'd. Non-static pod has no problem, but static seems to be affected by it.

### Kubernetes version

<details>

v1.22.1

</details>

### Cloud provider

<details>
Local Cluster
</details>

### OS version

<details>
</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by V0idk**
…ner status from container runtime.

#### What type of PR is this?

/kind bug

#### Which issue(s) this PR fixes:

Fixes #108537

#### Does this PR introduce a user-facing change?

```release-note
Fix static pod status is always Init:0/1 if unable to get init container status from container runtime.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

---
Before fix: 

image

After fix: 

image

```yaml
root@vm:~# kubectl get po  busybox-minikube  -o yaml
apiVersion: v1
kind: Pod
metadata:
....
  initContainerStatuses:
  - image: 192.168.6.100:5000/busybox:latest
    imageID: ""
    lastState: {}
    name: init1
    ready: true
    restartCount: 0
    started: false
    state:
      terminated:
        exitCode: 0
        finishedAt: null
        message: Unable to get init container status from container runtime and pod
          has been initialized, treat it as exited normally
        reason: Completed
        startedAt: null
```

### CODE REVIEW

**Comment by V0idk** [comment]
I have solved the test

image

**Comment by V0idk** [comment]
@bart0sh 
Check again plz

**Comment by pohly** [comment]
Does the release-note block have this format?

    ```release-note
    Fix static pod status is always Init:0/1 if unable to get init container status from container runtime.
    ```

File: pkg/kubelet/kubelet_pods.go [hunk 1]
```diff
@@ -2167,6 +2170,16 @@ func (kl *Kubelet) convertToAPIContainerStatuses(pod *v1.Pod, podStatus *kubecon
...
 			}
+			if s == nil && podHasInitialized && statuses[container.Name].State.Waiting != nil {
+				statuses[container.Name].State = v1.ContainerState{
+					Terminated: &v1.ContainerStateTerminated{
+						Reason:   "Completed",
+						Message:  "Unable to get init container status from container runtime and pod has been initialized, treat it as exited normally",
+						ExitCode: 0,
+					},
+				}
+				continue
```
Comment (pacoxu) [inline pkg/kubelet/kubelet_pods.go:2181]: Should we handle the new sidercar containers the same way? https://github.com/kubernetes/enhancements/issues/753

**Pull request closed without merging**